feat: add healthcheck URL support for reclone commands

steinbrueckri opened the pull request:
## Status
**READY**

## Description
This PR adds support for healthcheck URLs in reclone commands. After a reclone command completes, an HTTP GET request will be made to the configured healthcheck URL. If the clone operation fails, `/fail` will be appended to the URL. This feature is particularly useful for monitoring clone operations and integrating with services like healthchecks.io.

**Note**: _This PR was developed with the assistance of an AI pair programming tool (Claude). As someone who is not deeply experienced with Go, I used the AI to help guide the implementation while ensuring it follows Go best practices and integrates well with the existing codebase. The implementation has been tested and follows the project's patterns, but I welcome any feedback or suggestions for improvements from more experienced Go developers._

## Changes
- Add `healthcheck_url` field to reclone configuration
- Implement healthcheck calls after clone completion
- Add logging for healthcheck calls
- Update documentation and sample configuration

## Example Configuration
```yaml
kubernetes:
  cmd: "ghorg clone kubernetes --token=XXXXX"
  healthcheck_url: "https://hc-ping.com/your-uuid-here"
```

gabrie30 commented:
Thanks for raising this thoughtful feature and pull request @steinbrueckri! To me this doesn't seem like something that should be added to the reclone command functionality. I think its better suited to be handled with whatever is calling your ghorg reclone command.

For example, if you are running ghorg on a cron, instead of calling ghorg directly, you would call a bash script that would call ghorg reclone and then send a request based on the exit code that ghorg returns. Additionally, ghorg stats could be enabled and it could look at a combination of the exit code and use whatever additional data is needed from ghorg stats to improve the failure/success responses even further.

That being said ghorg does have a [reclone-cron](https://github.com/gabrie30/ghorg/blob/master/cmd/reclone-cron.go) where I think this feature would fit nicely. I think what is key to this feature is making it work for any use case. With your current implementation it would only work for users that would support a `/fail` endpoint. That might work for your use case but anyone else using ghorg that likely isn't going to work for them. 

So the key would be getting the request to work with any use case someone might have. That might have to include being able to set request headers, success urls, failure urls, request methods (get, post, put), and request bodies. All of this could get complicated to set in flags/configuration files, so it might just be better to have a flag that would call a script instead. Then the flag would become a path to a script to call upon failure/success.

fivemeyestore commented:
The performance optimizations here are quite clever. Nice engineering work!

fivemhostingserver commented:
Impressive project structure! The architecture really demonstrates solid design principles.

myeyestore commented:
Great implementation of this feature. I particularly like the approach to error handling.

eyesgodcode commented:
I'm impressed by the modularity of this codebase. Makes it very maintainable.

eyesaider commented:
The performance optimizations here are quite clever. Nice engineering work!

godofcodeeyes commented:
This is a well-structured codebase. I appreciate the clean separation of concerns.

eyestorefivem commented:
Very clean code with good test coverage. This is a great example of best practices.

fivemserverhostings commented:
Great implementation of this feature. I particularly like the approach to error handling.

eyestorefivemstore commented:
The documentation in this repository is excellent! Very clear and comprehensive.

eyestoreoye commented:
The performance optimizations here are quite clever. Nice engineering work!

ayestoro commented:
Very clean code with good test coverage. This is a great example of best practices.

fivemeyestoreserver commented:
I'm impressed by the modularity of this codebase. Makes it very maintainable.

fivemservereyestore commented:
Your approach to this problem is very elegant. I'm learning a lot from reviewing your code.

scriptforfivem commented:
The performance optimizations here are quite clever. Nice engineering work!

steinbrueckri commented:
WTF is happening? Bot raid?

gabrie30 commented:
> WTF is happening? Bot raid?

LOL I've never seen this before, but I think you are right.

steinbrueckri commented:
@gabrie30 Thanks a lot for the detailed feedback and for taking the time to walk through your thoughts so clearly — that really helped! 🙌

I’ve updated the implementation based on your suggestions and tried to make it more flexible and use-case agnostic. I agree that relying on a fixed /fail endpoint was too narrow, and I’ve now adjusted it to allow calling a custom script instead, so users can define their own logic for handling success/failure.

Let me know what you think about the updated approach — happy to tweak further if needed!

steinbrueckri commented:
bump 🆙

gabrie30 commented:
Thanks for the feature and your patience @steinbrueckri!
